alacritty/src/event.rs

```diff
@@ -314,12 +315,29 @@ impl<'a, N: Notify + 'a, T: EventListener> input::ActionContext<T> for ActionCon
             #[cfg(target_os = "freebsd")]
             let link_path = format!("/compat/linux/proc/{}/cwd", pid);
 
-            fs::read_link(link_path)
+            // Add the current working directory as parameter.
+            let args = fs::read_link(link_path)
                 .map(|path| vec!["--working-directory".into(), path])
-                .unwrap_or_default()
+                .unwrap_or_default();
+
+            args
         };
+
         #[cfg(not(unix))]
-        let args: Vec<String> = Vec::new();
+        let mut args: Vec<PathBuf> = Vec::new();
+
+        let working_directory_set = !args.is_empty();
+
+        // Reuse the arguments passed to Alacritty for the new instance.
+        while let Some(arg) = env_args.next() {
```

Comment:
I think we can just use `filter` method here.
`
        args.extend(
            env_args
                .filter(|x| !working_directory_set || x != "--working-directory")
                .map(|x| x.into()),
        );

`

Reply:
We can't, look at what the code does. It doesn't just filter one element.

Reply:
Alright, So this will be a place to filter out other arguments as well?

Reply:
Oh right, my bad